lib/addons/fluent-bit/index.ts

```diff
@@ -0,0 +1,117 @@
+import * as eks from '@aws-cdk/aws-eks';
+import * as es from '@aws-cdk/aws-elasticsearch';
+import { Effect, PolicyStatement } from '@aws-cdk/aws-iam';
+
+import { Constants } from "..";
+import { ClusterInfo } from "../../spi/types";
+import { Team } from "../../spi/team-contracts";
+import { ClusterAddOn } from "../../spi/addon-contracts"
+import { createNamespace } from "../../utils/namespace";
+
+/**
+ * Configuration options for the FluentBit add-on.
+ */
+export interface FluentBitAddOnProps {
```

Comment:
Thinking more about the design, it sounds like the plugin could support not just other destinations, but also multiple destinations at the same time. So purely for design perspective we could create an interface of destination providers and allow customers to pass those in.

It also sounds like the service account should be created by us upfront and each destination may add specific policies to it. 

this will make it more complex, while usage convenience will remain roughly the same. Could be deferred for pragmatic reasons, but sounds like a one way door once it is out there.